### PR TITLE
chore: disable buy/swap if currency not supported

### PR DESCRIPTION
**Pull request by dgacevic**
https://bayes.atlassian.net/browse/RWPD-484

### CODE REVIEW

File: brd-ios/breadwallet/Helpers/SupportedCurrenciesManager.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+//  Created by Dino Gacevic on 24/04/2023.
+//  Copyright © 2023 RockWallet, LLC. All rights reserved.
+//
+//  See the LICENSE file at the project root for license information.
+//
+
+import Foundation
+
+// TODO: Replace direct usage of supported currencies worker across app
+class SupportedCurrenciesManager {
```
[the remark this answers is not in the document]
Comment (dgacevic) [reply brd-ios/breadwallet/Helpers/SupportedCurrenciesManager.swift:14]: In this case it's being called once and then the result would be reused, so there is no need for a BE call everytime you need to access supported currencies.

File: brd-ios/breadwallet/Helpers/SupportedCurrenciesManager.swift [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    var supportedCurrencies: [SupportedCurrency] = []
+    
+    func getSupportedCurrencies() {
+        SupportedCurrenciesWorker().execute { [weak self] result in
+            switch result {
+            case .success(let currencies):
+                self?.supportedCurrencies = currencies ?? []
+                
+            case .failure(let error):
+                // TODO: Add some other form of error handling
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/Helpers/SupportedCurrenciesManager.swift:26]: you added error but did not remove the comment

File: brd-ios/breadwallet/Helpers/SupportedCurrenciesManager.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (dgacevic) [reply brd-ios/breadwallet/Helpers/SupportedCurrenciesManager.swift:14]: Tested it now without this manager, it is much better than calling supported currencies worker from 10 different entry points for asset details controller. I would keep this solution for now.

File: brd-ios/breadwallet/src/Redesign/Scenes/Buy/Main/BuyInteractor.swift [hunk 3]
```diff
@@ -26,29 +26,20 @@ class BuyInteractor: NSObject, Interactor, BuyViewActions {
...
+        let currencies = SupportedCurrenciesManager.shared.supportedCurrencies
+        
+        guard !currencies.isEmpty else { return }
+        
+        dataStore?.supportedCurrencies = currencies
+        dataStore?.currencies = dataStore?.currencies.filter { cur in currencies.map { $0.code }.contains(cur.code) } ?? []
+        
+        presenter?.presentData(actionResponse: .init(item: Models.Item(type: dataStore?.paymentMethod,
+                                                                             achEnabled: UserManager.shared.profile?.kycAccessRights.hasAchAccess)))
+        presenter?.presentAssets(actionResponse: .init(amount: dataStore?.toAmount,
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/Buy/Main/BuyInteractor.swift:38]: indentation issue

**Review by kenanmamedoff — APPROVED** [review]

**Pull request merged**